controller/lib/core/comms.cpp

```diff
@@ -204,54 +37,77 @@ void comms_sendControllerStatus(ControllerStatus controller_status) {
 // TODO add CRC to whole packet
 
 // TODO run this via DMA to free up resources for control loops
-static void process_tx() {
-  if (output_buffer_ready) {
-    for (uint16_t i = 0; i < tx_data_length; i++) {
-      uint16_t written = Hal.serialWrite(tx_buffer[i]);
-      if (1 != written) {
-        // TODO catch on fire
-      }
+static void process_tx(const ControllerStatus &controller_status) {
+  auto bytes_avail = Hal.serialBytesAvailableForWrite();
```

Comment:
Both conditions below require this > 0, maybe say "if (<= 0) return"?

Reply:
Done!